v2/pkg/controller/mpi_job_controller.go

```diff
@@ -459,6 +459,13 @@ func (c *MPIJobController) syncHandler(key string) error {
 		}
 	}
 
+	if len(mpiJob.Status.Conditions) == 0 {
```

Comment:
Problems with the previous approach:

- We were modifying an object in the informer cache, instead of modifying a copy. That can lead to undefined behavior.
- If the first Job status update failed, we might never add the condition again.
- If the job controller restarts, we were resetting the condition timestamp and incrementing the mpiJobscreateCount metric for all existing Jobs. When the controller restarts, every job produces an Add event.